selftests/functional/plugin/test_jsonresult.py

```diff
@@ -29,3 +29,14 @@ def test_fail_reason(self):
             test_data = data['tests'].pop()
             self.assertEqual('This test is supposed to fail',
                              test_data['fail_reason'])
+
+    def test_tags_in_result(self):
+        cmd_line = (f'{AVOCADO} run examples/tests/failtest.py '
+                    f'--job-results-dir {self.tmpdir.name} --disable-sysinfo')
+        process.run(cmd_line, ignore_status=True)
+        json_path = path.join(self.tmpdir.name, 'latest', 'results.json')
+        with open(json_path, 'r', encoding='utf-8') as json_file:
+            data = json.load(json_file)
+            test_data = data['tests'].pop()
```

Comment:
This is just a matter of style, of course, but to me the call to `pop()` seems to indicate that the `data['tests']` needs to have that item removed.

As a suggestion, I'd simply do:

```diff
diff --git a/selftests/functional/plugin/test_jsonresult.py b/selftests/functional/plugin/test_jsonresult.py
index 0c400ccf3..12942de4d 100644
--- a/selftests/functional/plugin/test_jsonresult.py
+++ b/selftests/functional/plugin/test_jsonresult.py
@@ -37,6 +37,5 @@ class JsonResultTest(TestCaseTmpDir):
         json_path = path.join(self.tmpdir.name, 'latest', 'results.json')
         with open(json_path, 'r', encoding='utf-8') as json_file:
             data = json.load(json_file)
-            test_data = data['tests'].pop()
             self.assertEqual({'failure_expected': None},
-                             test_data['tags'])
+                             data['tests'][0]['tags'])
```

Reply:
sure, I can change this.